lib/galaxy/config.py

```diff
@@ -685,6 +681,27 @@ def __init__(self, **kwargs):
 
         self.containers_conf = parse_containers_config(self.containers_config_file)
 
+        # Compliance/Policy variables
+        self.redact_username_during_deletion = False
+        self.redact_email_during_deletion = False
+        self.redact_ip_address = False
+        self.redact_username_in_logs = False
+        self.redact_email_in_job_name = False
+        self.redact_user_details_in_bugreport = False
+        # GDPR compliance mode changes values on a number of variables. Other
+        # policies could change (non)overlapping subsets of these variables.
+        self.gdpr_compliance_mode = string_as_bool(kwargs.get("gdpr_compliance_mode", False))
+        if self.gdpr_compliance_mode:
+            self.expose_user_name = False
+            self.expose_user_email = False
+
+            self.redact_username_during_deletion = True
+            self.redact_email_during_deletion = True
+            self.redact_ip_address = True
+            self.redact_username_in_logs = True
+            self.redact_email_in_job_name = True
+            self.redact_user_details_in_bugreport = True
```

Comment:
Thanks for making this change, I do think this is better. I do think I feel like this too granular now though? I would have just done redact_user_email and redact_username instead of specifying the context also. I also would have read the defaults above that are all False from kwargs so they are usable in other contexts. The good thing about not reading it from kwargs and not making it quite usable in other contexts though is that we have some time to find the right level of granularity without worrying about config interface changes.

Reply:
Sure, probably went too far in one direction. But yes, exactly what was great about your request originally, we leave one config option for public interface and can do any amount of refactoring later if it proves better.